compiler/rustc_arena/src/lib.rs

```diff
@@ -556,8 +557,19 @@ struct DropType {
     obj: *mut u8,
 }
 
-unsafe fn drop_for_type<T>(to_drop: *mut u8) {
-    std::ptr::drop_in_place(to_drop as *mut T)
+// SAFETY: we require `T: Send` before type-erasing into `DropType`.
+#[cfg(parallel_compiler)]
```

Comment:
Why does this need to be gated on parallel compiler? We always enforce the bound, right?

Reply:
In non-parallel builds, `sync::Send` is a _safe_ dummy trait that is already implemented for all `T: ?Sized`. An `unsafe impl` wouldn't work, and it's unneeded anyway. And we don't want to always require and implement the real `Send`, because conditional types like `Lrc` wouldn't be accepted in non-parallel builds.

Reply:
Ah, right, for some reason I thought in non-parallel builds it re-exported the regular Send, but that doesn't actually make sense, ok, thanks for explaining!